tests/test_part_image.py

```diff
@@ -124,15 +124,29 @@ def test_partition_alignment(self, bitbake_path, bitbake_variables, latest_part_
         assert(parts_start[2] == parts_end[1])
         assert(parts_start[3] == parts_end[2])
 
+        if bitbake_variables["MENDER_DEVICE_TYPE"] == "qemux86_64":
+            # For both data and swap partitons a default offset is added.
+            # Offset value is equal to alignment.
+            assert(parts_start[4] == parts_start[3] + alignment)
+            assert(parts_start[5] == parts_end[4] + alignment)
+
         # Partitions should extend for their size rounded up to alignment.
         # No set size for Rootfs partitions, so cannot check them.
         # Boot partition.
         assert(parts_end[0] == parts_start[0] + align_up(boot_part_size, alignment))
         # Data partition.
-        assert(parts_end[3] == parts_start[3] + align_up(data_part_size, alignment))
+        if bitbake_variables["MENDER_DEVICE_TYPE"] == "qemux86_64":
+            data_part_index = 4
+        else:
+            data_part_index = 3
```

Comment:
You can find this out also by doing `if len(parts_start) >= 5`.

Reply:
Here is slightly different situation as I do not have info about number of partitions in the processed image. So 'test_device_type' & 'test_data_ownership' must be handled in a different way. Maybe I could add something to DISTRO_FEATURES?

Reply:
I'm not sure I understand. This is the same function as the previous comment addresses, and you have the information there. If there are five partitions or more, you know that the data partition has to be at index 4, not 3.

Reply:
@kacf Not exactly, unless I'm missing something. _test_partition_alignment_ uses sub-process call to 'fdisk' to get the number of partitons. _test_data_ownershio_ & _test_device_type_ don't use the sub-process; maybe I should extract these few lines to a separate method and re-use it in two mentioned tests too?

Reply:
Oh, I think you're actually talking about the comments further down, that are in different functions. Then I see your point. Your solution sounds good, to extract the lines into a separate method and call it from all three test functions.

Reply:
Ok, thanks.